consensus/validation_fuzz.go

```diff
@@ -3,21 +3,54 @@
 package consensus
 
 import (
-	"bytes"
+	"crypto/ed25519"
 	"go.sia.tech/core/types"
-	"io"
 )
 
 func Fuzz(data []byte) int {
-	d := types.NewDecoder(io.LimitedReader{bytes.NewReader(data), 1e7})
+	d := types.NewBufDecoder(data)
 
 	var txn types.Transaction
 	txn.DecodeFrom(d)
 	if err := d.Err(); err != nil {
 		return -1
 	}
 
+	var merkleTxn types.Transaction
+	for _, input := range txn.SiacoinInputs {
+		merkleTxn.SiacoinOutputs = append(merkleTxn.SiacoinOutputs, types.SiacoinOutput{
+			Value:   input.Parent.Value,
+			Address: input.Parent.Address,
+		})
+	}
+	for _, input := range txn.SiafundInputs {
+		merkleTxn.SiafundOutputs = append(merkleTxn.SiafundOutputs, types.SiafundOutput{
+			Value:   input.Parent.Value,
+			Address: input.Parent.Address,
+		})
+	}
+
 	var ctx ValidationContext
+	au := ApplyBlock(ctx, types.Block{
+		Transactions: []types.Transaction{merkleTxn},
+	})
+	for i, elem := range au.NewSiacoinElements {
+		txn.SiacoinInputs[i].Parent = elem
+	}
+	for i, elem := range au.NewSiafundElements {
+		txn.SiafundInputs[i].Parent = elem
+	}
+	// use first 32 bytes of transaction data as seed for keys to make it deterministic
+	privkey := ed25519.NewKeyFromSeed(data[:32])
```

Comment:
for the signatures to be valid, they need to come from the key that controls the input. So you'll want to derive this key first, then use it to set `input.PublicKey`, and set `output.Address` equal to `types.StandardAddress(pubkey)`.